### PR TITLE
Add new plot_waveform code

### PR DESCRIPTION
**Pull request by spxiwh**
Add a code to make plots of template waveforms. This requires a new function in waveform.py to return h+ and hx (which I'll post a PR for soon). An example output is this particularly bad example of a NSBH SpinTaylorF2 waveform:

http://galahad.aei.mpg.de/~spxiwh/H1-PLOT_WAVEFORM_P1_0-1126051217-220400.png

which seems to suggest that SpinTaylorF2 has similar issues to IMRPhenomP ... but more on that soon.

### CODE REVIEW

**Comment by spxiwh** [comment]
Updated version of code, with inset on merger in time domain:

http://galahad.aei.mpg.de/~spxiwh/H1-PLOT_WAVEFORM_P1_1-1126051217-220400.png

File: bin/plotting/pycbc_plot_waveform [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+parser.add_argument("--spin1y", type=float, default=0,
+                  help="Non-aligned spin of the first component object. ")
+parser.add_argument("--spin2x", type=float, default=0,
+                  help="Non-aligned spin of the second component object. ")
+parser.add_argument("--spin2y", type=float, default=0,
+                  help="Non-aligned spin of the second component object. ")
+parser.add_argument("--inclination", type=float, default=0,
+                  help="Non-aligned spin of the second component object. ")
+parser.add_argument("--u-val", type=float, default=0,
+                  help="Ratio between h_+ and h_x according to .... . ")
```
Comment (cdcapano) [inline bin/plotting/pycbc_plot_waveform]: What's u-val? Can you give a better help message? At least say when you can ignore this.

**Comment by spxiwh** [comment]
Hi Collin. I've updated the help message as regards u_val. This is the third parameter that SkyMax inspiral maximizes over (in addition to amplitude and phase).

File: bin/plotting/pycbc_plot_waveform [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+                  help="Non-aligned spin of the second component object. ")
+parser.add_argument("--spin2y", type=float, default=0,
+                  help="Non-aligned spin of the second component object. ")
+parser.add_argument("--inclination", type=float, default=0,
+                  help="Non-aligned spin of the second component object. ")
+parser.add_argument("--u-val", type=float, default=0,
+                  help="Ratio between h_+ and h_x according to 
+                        h(t) = h_+ * u_val + h_x. Only needed in the case of
+                        waveforms where h_+ and h_x are not related by a 
+                        simple phase shift.")
```
Comment (cdcapano) [inline bin/plotting/pycbc_plot_waveform]: Don't you need ending/starting quotes at the end/beginning of each line?

**Comment by spxiwh** [comment]
Fixed, and rebased against master to resolve merge conflict.

**Pull request merged**